From 65f2a8665677f3a1489f951375e42f936ac654ec Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Tue, 30 May 2023 00:02:55 +0200
Subject: [PATCH 404/406] Revert "Bluetooth: Add new quirk for broken local ext
 features max_page"

This reverts commit 6fed65c5068b79600eb49c50f59d482c67ee37be.
---
 include/net/bluetooth/hci.h | 7 -------
 net/bluetooth/hci_event.c   | 4 +---
 2 files changed, 1 insertion(+), 10 deletions(-)

diff --git a/include/net/bluetooth/hci.h b/include/net/bluetooth/hci.h
index 6c2d5f36b600..400f8a7d0c3f 100644
--- a/include/net/bluetooth/hci.h
+++ b/include/net/bluetooth/hci.h
@@ -294,13 +294,6 @@ enum {
 	 * during the hdev->setup vendor callback.
 	 */
 	HCI_QUIRK_BROKEN_MWS_TRANSPORT_CONFIG,
-
-	/* When this quirk is set, max_page for local extended features
-	 * is set to 1, even if controller reports higher number. Some
-	 * controllers (e.g. RTL8723CS) report more pages, but they
-	 * don't actually support features declared there.
-	 */
-	HCI_QUIRK_BROKEN_LOCAL_EXT_FTR_MAX_PAGE,
 };
 
 /* HCI device flags */
diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index c4b3969058a5..e87c928c9e17 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -886,9 +886,7 @@ static u8 hci_cc_read_local_ext_features(struct hci_dev *hdev, void *data,
 	if (rp->status)
 		return rp->status;
 
-	if (!test_bit(HCI_QUIRK_BROKEN_LOCAL_EXT_FTR_MAX_PAGE,
-		      &hdev->quirks) &&
-	    hdev->max_page < rp->max_page)
+	if (hdev->max_page < rp->max_page)
 		hdev->max_page = rp->max_page;
 
 	if (rp->page < HCI_MAX_PAGES)
-- 
2.34.1

